### ISSUE TITLE
Support `aux` channels on the ECM

### ISSUE THREAD

**Issue by sdwilsh** [sdwilsh/brultech-serial2mqtt#246]
This would would entail the following things:
1) Teach `ChannelConfig` about `aux` channels.  See https://github.com/sdwilsh/brultech-serial2mqtt/blob/main/brultech_serial2mqtt/config/config_device.py
2) Teach the channels code to utilize the new config as well as how to pull the data from the packets.  See https://github.com/sdwilsh/brultech-serial2mqtt/blob/main/brultech_serial2mqtt/device/channel.py

Plus additional test coverage for all of this, which is especially important since I don't have an ECM to test with.

**Comment by 42Network** [comment]
Thanks for these details. What is the typical GitHub etiquette? If I want to work on this ECM Aux issue, can it be assigned to me? I think that would mean I need additional permissions. But even if I don't have additional permissions, I suppose I can learn how to submit PRs for you to approve and merge?

Additionally, I have noticed you guys don't use the GitHub labels, such as "Bug", "Enhancement", etc. Is that a feature in GitHub that is controlled at the repository or project level?

**Comment by sdwilsh** [comment]
I can assign it to you, but this project has so little activity it doesn't really matter 😅

The labels things is kinda the same.  If we were inundated with issues, is start labeling stuff, but we have so few it's just extra work!

> I think that would mean I need additional permissions. But even if I don't have additional permissions, I suppose I can learn how to submit PRs for you to approve and merge?

You'll create your own fork and submit PRs from that.  Check out https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Comment by 42Network** [comment]
Hello Shawn,

I am (very slowly) working on making your package support my ECM-1240 Aux channels. I am confused by the following bit of code from config_device.py. I am getting used to both the idea of Python typing and list comprehensions, which all seem very powerful, but this code is somewhat opaque to me still. Does it imply that it only understands polarized channels? On an ECM-1240, only the 2 main channels are polarized. Aux 1-5 are not polarized. The list called polarized_channels seems to include only the channel numbers that are of ChannelType MAIN or SOLAR. self._channels seems to just be a copy of polarized_channels. How are non-polarazied channels taken into account?

```
class ChannelsConfig:
    def __init__(self, channels_config: List[Dict[str, Any]]):
        polarized_channels = ChannelsConfig._get_polarized_channels(channels_config)
        self._channels: dict[int, ChannelConfig] = {
            c["number"]: ChannelConfig(c, c["number"] in polarized_channels)
            for c in channels_config
        }
```

Additionally, do you think ChannelType of NORMAL = 1 (line 20) is ok to use for Aux channels, or should I introduce an additional enum for Aux?

```
class ChannelType(Enum):
    NORMAL = 1
    MAIN = 2
    SOLAR_DOWNSTREAM_MAIN = 3
    SOLAR_UPSTREAM_MAIN = 4

```
Thanks/Nathan

**Comment by sdwilsh** [comment]
> Does it imply that it only understands polarized channels?

No, if you look at `_get_polarized_channels`, we're only adding specific channels to it (main and solar ones).  This is just some magic I added in order to make Home Assistant's energy dashboard work well.  So if the channel config is of type `ChannelType.MAIN`, `ChannelType.SOLAR_DOWNSTREAM_MAIN`, or `ChannelType.SOLAR_UPSTREAM_MAIN`, it would be considered as a polarized channel here.

For the ECM, if any of the `aux` channels have those channel types, it's probably worth raising a config error (which we can also write a test for) if they cannot be supported as polarized on the device.

> The list called polarized_channels seems to include only the channel numbers that are of ChannelType MAIN or SOLAR. self._channels seems to just be a copy of polarized_channels. How are non-polarazied channels taken into account?

This might be an artifact of the ECM and it's limited channels.  `self._channels` will be a dictionary of channel numbers to that channel's config, and it is defined by the `yaml` file.  For instance, the channel config I have for my home is [here](https://github.com/sdwilsh/ansible-playbooks/blob/main/roles/docker-brultech-serial2mqtt/tasks/main.yml#L34-L106), and so my `polarized_channels` just contains two (`17` and `25`).


It might be easiest if you make a draft PR of what you have so far, and I can help guide you to getting this implemented, but that's up to you.  I'll make it a point to check my GitHub email a bit more regularly so you hopefully only have a single day of turn-around time :)
